src/common/config_parser.cpp

```diff
@@ -679,16 +696,25 @@ void ConfigParser::addOptionsTranslation(cli::CLIWrapper& cli) {
       "Keep the output segmented into SentencePiece subwords");
 #endif
 
-  addSuboptionsInputLength(cli);
-  addSuboptionsTSV(cli);
-  addSuboptionsDevices(cli);
-  addSuboptionsBatching(cli);
+  if(mode_ != cli::mode::selfadaptive) {
+    addSuboptionsInputLength(cli);
+    addSuboptionsTSV(cli);
+    addSuboptionsDevices(cli);
+    addSuboptionsBatching(cli);
+  } else {
+    cli.add<size_t>("--max-length-translate",
+        "Maximum input sentence length for translation",
+        1000);
+  }
 
-  cli.add<bool>("--fp16",
-      "Shortcut for mixed precision inference with float16, corresponds to: --precision float16");
-  cli.add<std::vector<std::string>>("--precision",
-      "Mixed precision for inference, set parameter type in expression graph",
-      {"float32"});
+  // for self-adaptive mode vocabs are already added via the training options
```

Comment:
The comment seems copied from another place and not relevant here, please update.

Reply:
Removed